### PR TITLE
delete many unused exported functions in pkg directory

### PR DESCRIPTION
**Pull request by Abirdcfly**
Signed-off-by: Abirdcfly <fp544037857@gmail.com>

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

part of https://github.com/kubernetes/kubernetes/issues/111666

#### Special notes for your reviewer:

This pr only removes the exposed functions that are not used by k8s (especially not by _test.go or other tests) and are located in the pkg directory

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by Abirdcfly** [comment]
/uncc @bobbypage @alculquicondor @caesarxuchao 

Let me verify it first

File: pkg/scheduler/framework/plugins/examples/multipoint/multipoint.go [hunk 1]
```diff
@@ -84,8 +83,3 @@ func (mc CommunicatingPlugin) PreBind(ctx context.Context, state *framework.Cycl
 	}
 	return nil
 }
-
-// New initializes a new plugin and returns it.
-func New(_ *runtime.Unknown, _ framework.Handle) (framework.Plugin, error) {
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/examples/multipoint/multipoint.go:89]: This is an example, so we should keep it.

**Review by alculquicondor — COMMENTED** [review]
Most of scheduling changes lgtm.

File: pkg/scheduler/framework/plugins/examples/multipoint/multipoint.go (hunk 1 above)
Comment (Abirdcfly) [reply pkg/scheduler/framework/plugins/examples/multipoint/multipoint.go]: Done

Comment (Abirdcfly) [reply pkg/scheduler/framework/plugins/examples/multipoint/multipoint.go]: revert all change in `pkg/scheduler/framework/plugins/examples`

**Review by alculquicondor — COMMENTED** [review]
/approve

for scheduler

File: pkg/apis/admission/register.go [hunk 2]
```diff
@@ -27,16 +27,6 @@ const GroupName = "admission.k8s.io"
 // SchemeGroupVersion is group version used to register these objects
 var SchemeGroupVersion = schema.GroupVersion{Group: GroupName, Version: runtime.APIVersionInternal}
 
-// Kind takes an unqualified kind and returns a Group qualified GroupKind
```
Comment (Abirdcfly) [inline pkg/apis/admission/register.go:30]: I would be happy to hear suggestions from the community on whether the large number of similar **unused exposed** `Kind` and `Resource` functions should be removed or retained

File: pkg/kubelet/container/resize.go [hunk 3]
```diff
@@ -1,42 +0,0 @@
-/*
```
Comment (Abirdcfly) [inline pkg/kubelet/container/resize.go:1]: I would be happy to hear suggestions from the community on whether the current file (package) should be deleted or retained.

File: pkg/kubelet/kubeletconfig/util/panic/panic.go [hunk 4]
```diff
@@ -1,36 +0,0 @@
-/*
```
Comment (Abirdcfly) [inline pkg/kubelet/kubeletconfig/util/panic/panic.go:1]: I would be happy to hear suggestions from the community on whether the current file (package) should be deleted or retained.

File: pkg/kubelet/kubeletconfig/util/equal/equal.go [hunk 5]
```diff
@@ -1,24 +0,0 @@
-/*
```
Comment (Abirdcfly) [inline pkg/kubelet/kubeletconfig/util/equal/equal.go:1]: I would be happy to hear suggestions from the community on whether the current file (package) should be deleted or retained.

File: pkg/util/bandwidth/OWNERS [hunk 6]
```diff
@@ -1,8 +0,0 @@
-# See the OWNERS docs at https://go.k8s.io/owners
```
Comment (Abirdcfly) [inline pkg/util/bandwidth/OWNERS:1]: I would be happy to hear suggestions from the community on whether the current file (package) should be deleted or retained.

@kubernetes/sig-network-approvers
@andrewsykim @aojea @bowei @caseydavenport @danwinship @dcbw @freehan @khenidak @mrhohn @robscott @thockin

File: pkg/util/maps/string.go [hunk 7]
```diff
@@ -1,29 +0,0 @@
-/*
```
Comment (Abirdcfly) [inline pkg/util/maps/string.go:1]: I would be happy to hear suggestions from the community on whether the current file (package) should be deleted or retained.

**Comment by Abirdcfly** [comment]
/cc @bobbypage  @caesarxuchao
/assign @thockin

(Restore @k8s-ci-robot's original assignment)

**Comment by BenTheElder** [comment]
This is a pretty large diff, can we split this up more granularly?

E.G. one PR might be removing the unused Kind and Resource methods.

This also makes getting approval easier because e.g. a PR cleaning up `pkg/util/bandwidth`can get sig-network approvers and each subset doesn't need _all_ of those people subscribed.

File: pkg/util/bandwidth/OWNERS (hunk 6 above)
Comment (khenidak) [reply pkg/util/bandwidth/OWNERS:1]: it should remain. Thank you.

File: pkg/util/bandwidth/doc.go [hunk 8]
```diff
@@ -1,18 +0,0 @@
-/*
```
Comment (khenidak) [inline pkg/util/bandwidth/doc.go:1]: please don't delete doc or owners files

Comment (khenidak) [reply pkg/util/bandwidth/doc.go:1]: oh! you removed the entire doc? the entire package is not used anywhere?

**Comment by khenidak** [comment]
/sig api-machinery
/sig network
/lgtm

The changes on network seems seems ok. Surprised to see tests passing with such a massive removal.

File: pkg/apis/flowcontrol/validation/validation.go [hunk 9]
```diff
@@ -354,11 +354,6 @@ func ValidatePriorityLevelConfiguration(pl *flowcontrol.PriorityLevelConfigurati
 	return allErrs
 }
 
-// ValidatePriorityLevelConfigurationUpdate validates the update of priority-level-configuration.
-func ValidatePriorityLevelConfigurationUpdate(old, pl *flowcontrol.PriorityLevelConfiguration) field.ErrorList {
```
Comment (deads2k) [inline pkg/apis/flowcontrol/validation/validation.go:358]: huh, weird that this isn't called, seems like a bug

Comment (Abirdcfly) [reply pkg/apis/flowcontrol/validation/validation.go:358]: [`66fc0d7` (#112306)](https://github.com/kubernetes/kubernetes/pull/112306/commits/66fc0d703794f309c9715028d3b63f64c281a5fd#r978608952)
look like it was removed 19 days ago.

**Comment by thockin** [comment]
I agree a bunch of small PRs would be better

**Comment by thockin** [comment]
@Abirdcfly are you coming back to this?

**Comment by Abirdcfly** [comment]
> @Abirdcfly are you coming back to this?

Sorry, I don't have time to continue this work at the moment, feel free to cut this commit if it would be helpful to anyone continuing this work.

**Pull request closed without merging**